### PR TITLE
Salvador development Step #4

### PR DESCRIPTION
**Pull request by sroblesher**
Stores comments of user

### CODE REVIEW

File: walkthroughs/week-3-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     // Convert the server stats to JSON
     ServerStats serverStats = new ServerStats(startTime, currentTime, maxMemory, usedMemory);
-    String json = convertToJson(serverStats);
+    String json = convertToJsonUsingGson(serverStats);
```
Comment (wait321) [inline walkthroughs/week-3-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java:40]: No need to commit changes when playing around with the examples. Revert this change.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -30,18 +30,29 @@
...
     response.getWriter().print(gson.toJson(messages));
     response.getWriter().flush();
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = getParameter(request, "text-comment", "");
+
+    messages.add(text);
+    response.setContentType("text/html;");
```
[the remark this answers is not in the document]
Comment (sroblesher) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: Which lines should I delete?

[the remark this answers is not in the document]
Comment (wait321) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: 46 and 47

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -30,18 +30,27 @@
...
     response.setContentType("application/json");
     response.setCharacterEncoding("UTF-8");
     response.getWriter().print(gson.toJson(messages));
     response.getWriter().flush();
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = getParameter(request, "text-comment", "");
+
```
Comment (aminry) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: nit: Please remove empty line

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -30,18 +30,27 @@
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json");
     response.setCharacterEncoding("UTF-8");
     response.getWriter().print(gson.toJson(messages));
     response.getWriter().flush();
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = getParameter(request, "text-comment", "");
```
Comment (aminry) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Should you check if the text is empty, and not add it to the list?

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -12,12 +12,20 @@ <h1>Salvador's Portfolio</h1>
...
+      <p>Display previous comments:</p>
+      <button onclick="displayComments()">Click me!</button>
+      <ol type="I" id="comments-container"></ol>
+
+      <form action="/data" method="POST" id="commentForm">
+        <p>Comment something...</p>
+        <textarea name="text-comment">Type here...</textarea>
+      </form>
+
+      <button type="submit" form="commentForm">Submit</button>
```
[the remark this answers is not in the document]
Comment (wait321) [reply portfolio/src/main/webapp/index.html:28]: This is important. The submit button finds the enclosing `<form>` tags. Right now, it's outside of that, so it will never submit and nothing will submit the comment in the textarea.

[the remark this answers is not in the document]
Comment (sroblesher) [reply portfolio/src/main/webapp/index.html:28]: I put a form="commentForm" inside the button tag, I thought that helped because it works

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (sroblesher) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Doesn't the getParameter function do that?
Or should I create an if statement to not add it to the list?

Comment (sroblesher) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: OH nvm I see the mistake
